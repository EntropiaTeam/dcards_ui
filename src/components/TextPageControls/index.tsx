import React, { FC } from 'react';
import {
  makeStyles, Box, Button
} from '@material-ui/core';
import { useHistory, useRouteMatch } from 'react-router-dom';
import { ZoomInIcon, ZoomOutIcon, EditIcon } from '../UI/Icons';
import { RoutePath } from '../../enums/Routes';

const useStyles = makeStyles({
  controlButtons: {
    position: 'fixed',
    top: '50%',
    transform: 'translate(-10px, -50%)',
    width: '40px',
    right: '0px',
    '&>button': {
      minWidth: '40px',
      width: '40px',
      height: '40px',
      padding: '0px',
      display: 'flex',
      justifyContent: 'center',
      alignItems: 'center',
      background: '#F6F6F6',
      marginBottom: '20px',
      '&:hover': {
        background: '#F6F6F6'
      },
      '&:last-child': {
        marginBottom: '0px'
      }
    }
  }
});

const TextPageControls: FC = () => {
  const classes = useStyles();
  const textEditorCard = document.getElementById('textEditorCard');
  const history = useHistory();
  const textEditorRouteMatch = useRouteMatch(RoutePath.TextEditor);

  const editMessage = () => {
    history.push(`${textEditorRouteMatch!.url}/mobile`);
  };

  const zoomIn = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();
    if (textEditorCard) {
      textEditorCard.style.transform = 'scale(1.2)';
    }
  };

  const zoomOut = (e: React.MouseEvent<HTMLButtonElement>) => {
    e.stopPropagation();
    if (textEditorCard) {
      textEditorCard.style.transform = 'scale(0.8)';
    }
  };

  return (
    <Box className={classes.controlButtons}>
      <Button color="primary" onClick={editMessage}><EditIcon /></Button>
      <Button color="primary" onClick={zoomIn}><ZoomInIcon /></Button>
      <Button color="primary" onClick={zoomOut}><ZoomOutIcon /></Button>
    </Box>
  );
};

export default TextPageControls;
