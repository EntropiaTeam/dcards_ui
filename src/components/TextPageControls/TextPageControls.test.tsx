import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import TextPageControls from '.';

const mockHistoryPush = jest.fn();

interface ReactRouterDom {
  useHistory: () => void;
  useRouteMatch: () => void;
}

interface IntlMock {
  useTranslation: () => void;
}

interface TranslationMock {
  t: () => void;
}

jest.mock('react-i18next', (): IntlMock => ({
  useTranslation: (): TranslationMock => ({
    t: (key: string): string => key
  })
}));

jest.mock('react-router-dom', (): ReactRouterDom => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    push: mockHistoryPush
  }),
  useRouteMatch: jest.fn().mockReturnValue(true)
}));

describe('<TextPageControls />', () => {
  const props = {
  };

  const TextPageControlsComponent = renderer.create(<TextPageControls {...props} />).toJSON();
  const wrapper = mount(<TextPageControls {...props} />);

  it('should track history when edit button click', () => {
    act(() => {
      wrapper.find('button').first().simulate('click');
    });
    expect(mockHistoryPush).toHaveBeenCalledTimes(1);
  });

  it('should render TextPageControls component correctly', () => {
    expect(TextPageControlsComponent).toMatchSnapshot();
  });
});
