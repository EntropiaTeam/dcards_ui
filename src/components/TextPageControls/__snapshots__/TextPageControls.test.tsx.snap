// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextPageControls /> should render TextPageControls component correctly 1`] = `
<div
  className="MuiBox-root MuiBox-root-2 makeStyles-controlButtons-1"
>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="#F6F6F6"
          height="40"
          rx="2"
          width="40"
        />
        <path
          d="M16.4873 20.1569L27.094 9.84412C27.4861 9.46294 28.1116 9.46734 28.4983 9.85399L29.9233 11.279C30.3134 11.6692 30.3138 12.3016 29.9242 12.6923L19.4873 23.1567L16.4873 20.1569Z"
          fill="#C8102E"
        />
        <path
          d="M15.049 24.6934L16.4873 20.1569L19.4874 23.157L15.049 24.6934Z"
          fill="currentColor"
        />
        <path
          d="M19.5 12.5H11V29.5H28V20"
          stroke="currentColor"
          strokeWidth="2"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="40"
          rx="2"
          width="40"
        />
        <path
          d="M24.7143 22.8286L31 29.1143L29.1143 31L22.8286 24.7143V23.7211L22.4891 23.3691C21.056 24.6011 19.1954 25.3429 17.1714 25.3429C15.0042 25.3429 12.9258 24.4819 11.3934 22.9495C9.86092 21.4171 9 19.3386 9 17.1714C9 15.0042 9.86092 12.9258 11.3934 11.3934C12.9258 9.86092 15.0042 9 17.1714 9C19.3386 9 21.4171 9.86092 22.9495 11.3934C24.4819 12.9258 25.3429 15.0042 25.3429 17.1714C25.3429 19.1954 24.6011 21.056 23.3691 22.4891L23.7211 22.8286H24.7143ZM17.1714 22.8286C20.3143 22.8286 22.8286 20.3143 22.8286 17.1714C22.8286 14.0286 20.3143 11.5143 17.1714 11.5143C14.0286 11.5143 11.5143 14.0286 11.5143 17.1714C11.5143 20.3143 14.0286 22.8286 17.1714 22.8286ZM20.3143 17.8H17.8V20.3143H16.5429V17.8H14.0286V16.5429H16.5429V14.0286H17.8V16.5429H20.3143V17.8Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <svg
        fill="none"
        height="40"
        viewBox="0 0 40 40"
        width="40"
        xmlns="http://www.w3.org/2000/svg"
      >
        <rect
          fill="none"
          height="40"
          rx="2"
          width="40"
        />
        <path
          d="M24.7143 22.8286H23.7211L23.3691 22.4891C24.6011 21.056 25.3429 19.1954 25.3429 17.1714C25.3429 15.0042 24.4819 12.9258 22.9495 11.3934C21.4171 9.86092 19.3386 9 17.1714 9C15.0042 9 12.9258 9.86092 11.3934 11.3934C9.86092 12.9258 9 15.0042 9 17.1714C9 19.3386 9.86092 21.4171 11.3934 22.9495C12.9258 24.4819 15.0042 25.3429 17.1714 25.3429C19.1954 25.3429 21.056 24.6011 22.4891 23.3691L22.8286 23.7211V24.7143L29.1143 31L31 29.1143L24.7143 22.8286ZM17.1714 22.8286C14.0286 22.8286 11.5143 20.3143 11.5143 17.1714C11.5143 14.0286 14.0286 11.5143 17.1714 11.5143C20.3143 11.5143 22.8286 14.0286 22.8286 17.1714C22.8286 20.3143 20.3143 22.8286 17.1714 22.8286ZM14.0286 16.5429H20.3143V17.8H14.0286V16.5429Z"
          fill="currentColor"
        />
      </svg>
    </span>
  </button>
</div>
`;
